spec-tests/src/test/java/org/candlepin/spec/DistributorCapabilitySpecTest.java

```diff
@@ -0,0 +1,309 @@
+            .displayName(StringUtil.random("displayName"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("midas touch"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("telepathy"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("lightning speed"));
+        distVersion = client.distributorVersions().create(distVersion);
+        assertThat(distVersion.getId()).isNotNull();
+        assertThat(client.distributorVersions().getVersions("test-creation", null))
+            .hasSize(count + 1);
+        assertThat(distVersion.getCapabilities()).hasSize(3);
+    }
+
+    @Test
+    public void shouldAllowDistributorVersionUpdate() {
+        int count = client.distributorVersions().getVersions("test-update", null).size();
+        DistributorVersionDTO distVersion = new DistributorVersionDTO()
+            .name(StringUtil.random("test-update"))
+            .displayName(StringUtil.random("displayName"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("midas touch"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("telepathy"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("lightning speed"));
+        distVersion = client.distributorVersions().create(distVersion);
+        String distVersionId = distVersion.getId();
+
+        DistributorVersionDTO updateDistVersion = new DistributorVersionDTO()
+            .name(distVersion.getName())
+            .displayName(distVersion.getDisplayName())
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("midas touch"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("lightning speed"));
+        distVersion = client.distributorVersions().update(distVersionId, updateDistVersion);
+
+        assertThat(client.distributorVersions().getVersions("test-update", null))
+            .hasSize(count + 1);
+        assertThat(distVersion)
+            .returns(distVersionId, x -> x.getId())
+            .extracting(DistributorVersionDTO::getCapabilities)
+            .returns(2, Set::size);
+    }
+
+    @Test
+    public void shouldAssignConsumerCapabilitiesBasedOnDistributorVersionWhenCreating() {
+        OwnerDTO owner = ownerClient.createOwner(Owners.random());
+        String name = StringUtil.random("name");
+        DistributorVersionDTO distVersion = new DistributorVersionDTO()
+            .name(name)
+            .displayName(StringUtil.random("displayName"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("midas touch"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("telepathy"))
+            .addCapabilitiesItem(new DistributorVersionCapabilityDTO().name("lightning speed"));
+        client.distributorVersions().create(distVersion);
+        Map facts = Map.of("distributor_version", name);
```

Comment:
Optional: You can move this into the .fact() method so that it is all in one line since the facts variable is only being used in that one spot:
`.facts(Map.of("distributor_version", name))`. This could also be done in some other spots in this test file as well.